src/models/ddex_models.py

```diff
@@ -1,116 +1,134 @@
 from django.db import models
-from .admin_models import Order
-from .tags import (
-        IdTypeSet,
-        PartyRoleSet,
-        PartyTypeSet,
-        )
-
+from .user_models import Order
+from .tags import Tags
+import uuid
 
 """
-All DDEX related models are available here that includes
-Party, Song, Image, Album and Clip
-
+----------------------------------------------------------------
+All DDEX related models are available here that includes Party, Song, Image, Album and Clip
+----------------------------------------------------------------
 TODO:
-    - Requires tests
 """
 
 
-ID_TYPE_CHOICES = [(id_type.value, id_type.value) for id_type in IdTypeSet] # Creates choices for id type
-
-PARTY_ROLE_CHOICES = [(role.value, role.value) for role in PartyRoleSet] # Creates choices for party roles
-
+class Album(models.Model):
+    '''
+     TODO: make signal if album id_type is specified then all song have same id_type
+    '''
 
-class Party(models.Model):
-    """
-    Stores data for PartyList tag in DDEX xml
-    """
-    class Meta:
-        verbose_name_plural = "Parties"
-    TYPE_CHOICES = [
-            ('Artist', PartyTypeSet.artist.value),
-            ('Contributor', PartyTypeSet.contributor.value)
-            ]
-    # default value is set to Artist from PartyTypeSet Enum
-    party_type = models.CharField(choices=TYPE_CHOICES, max_length=250, default=PartyTypeSet.artist.value) # Determines the type of Party i.e. either Artist or Contributor
-    role = models.CharField(choices=PARTY_ROLE_CHOICES, max_length=150, default=PartyRoleSet.main_artist.value) # Need to add choices
-    name = models.CharField(max_length=250) # Full name of the party
-    order = models.ManyToManyField(Order)
-
-    def get_reference(self, name, record_label):
-        return f"P{self.name}_{name}_{record_label}"
+    id_type = models.CharField(
+        max_length=100, null=True, blank=True, choices=Tags.ID_TYPE_CHOICE)
+    order = models.ForeignKey(
+        Order, on_delete=models.CASCADE, null=True, blank=True)
 
-    def __str__(self):
-        return f"{self.party_type}: {self.name}"
+    title = models.CharField(max_length=300, null=True, blank=True)
+    band_or_artist_name = models.CharField(max_length=300)
+    released_date = models.DateField(null=True, blank=True)
+    released_year = models.IntegerField(null=True, blank=True)
 
+    record_label = models.CharField(max_length=500, null=True, blank=True)
 
-class Album(models.Model):
-    id_type = models.CharField(choices=ID_TYPE_CHOICES, max_length=100, default=IdTypeSet.icpn.value)
-    # Same as song
-    # Needs a seperate validator for different types of id types
-    id_value = models.CharField(max_length=100, unique=True, default='A0')
-    name = models.CharField(max_length=150) # Name of the album
-    order = models.ForeignKey(Order, on_delete=models.CASCADE)
+    created = models.DateTimeField(auto_now_add=True)
+    id = models.UUIDField(default=uuid.uuid4, unique=True,
```

Comment:
This is not required. Album and song, will have their own ISRC id that will be used to refer to it.